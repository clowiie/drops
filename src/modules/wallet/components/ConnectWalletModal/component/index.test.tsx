import React from 'react'

import { render } from '@testing-library/react'

import mockComponent from 'test/utils/mockComponent'

import * as ConnectWalletModalType from '.'

describe('ConnectWalletModal', () => {
  const ModalSpy = mockComponent('common/components/Modal')
  const ConnectWalletChoiceSpy = mockComponent(
    'modules/wallet/components/ConnectWalletChoice',
  )

  const useConnectWalletModalSpy = jest.fn()
  const connectWalletSpy = jest.fn()
  jest.doMock(
    'modules/wallet/hooks/useConnectWalletModal',
    () => useConnectWalletModalSpy,
  )

  const { default: ConnectWalletModal } =
    require('.') as typeof ConnectWalletModalType

  beforeEach(() => {
    useConnectWalletModalSpy.mockReturnValue({
      connectWallet: connectWalletSpy,
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container } = render(<ConnectWalletModal isOpen />)

      expect(container.firstChild).toMatchSnapshot()
      expect(ModalSpy).toBeCalledTimes(1)
      expect(ConnectWalletChoiceSpy).toBeCalledTimes(2)
      expect(ConnectWalletChoiceSpy).toBeCalledWith({
        connectWallet: connectWalletSpy,
        connector: {
          connectorId: 'metaMask',
          logo: 'meta-mask-logo.svg',
          name: 'MetaMask',
        },
      })
      expect(ConnectWalletChoiceSpy).toBeCalledWith({
        connectWallet: connectWalletSpy,
        connector: {
          connectorId: 'walletConnect',
          logo: 'wallet-connect-logo.svg',
          name: 'WalletConnect',
        },
      })
    })
  })
})
