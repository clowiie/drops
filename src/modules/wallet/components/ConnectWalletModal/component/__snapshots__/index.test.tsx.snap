// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectWalletModal snapshots should match snapshots 1`] = `
<div>
  <div
    class="p-4 flex justify-around"
  >
    <div />
    <div />
  </div>
</div>
`;
